src/content/2.4/code/agda/snippet01.agda

```diff
@@ -0,0 +1,2 @@
+data Reader a x : Set where
+  reader : (a → x) → Reader a x
```

Comment:
My initial intuition would be to go for:
```Agda
Reader' = {A X : Set} → (A → X)
```
but I like this one too 👍 

In my Scala exploration I tired both encoding ([type alias](https://github.com/lemastero/scala_typeclassopedia/blob/main/src/main/scala/educational/types/Reader.scala) and [data type](https://github.com/lemastero/scala_typeclassopedia/blob/main/src/main/scala/educational/data/Reader.scala) declaration)